Launching kcauto from "GAME START" screen causes crash -- Ghost equipment workaround filter

#### Environment details

* kcauto version (version shown on startup): **latest `develop `version**
* Operating system:  **Windows 11 64bits**

#### Issue variability

Issue is consistent.

#### Issue

Launching `kcauto` from "GAME START" screen causes it to crash, but launching it from inside the game does not.

<img width="702" height="953" alt="Image" src="https://github.com/user-attachments/assets/3bd3e7cd-8ea9-49ef-b268-e46f196ca5d5" />

#### Log
```
[2026-01-19 03:14:28] Initializing kcauto.
[2026-01-19 03:14:28] Config successfully loaded.
[2026-01-19 03:14:30] Expedition module enabled.
[2026-01-19 03:14:30] Fleet 1 activated.
[2026-01-19 03:14:30] PvP module enabled.
[2026-01-19 03:14:30] Quest module enabled.
[2026-01-19 03:14:30] Loading Quest priority data.
[2026-01-19 03:14:31] Combat module enabled.
[2026-01-19 03:14:31] Next sortie at 2026-01-19 03:14:31
[2026-01-19 03:14:31] Hooking into Chrome.
[2026-01-19 03:14:52] Connected to Chrome
[2026-01-19 03:14:52] Finding browser offset
[2026-01-19 03:14:53] Browser offset found at X: 0, Y: 81
[2026-01-19 03:14:53] Initializing kcauto v3.1.3-beta in command-line mode with the 'config_cui' config file.
[2026-01-19 03:14:56] Starting kancolle from splash screen.
[2026-01-19 03:15:01] Fleet 2 activated.
[2026-01-19 03:15:01] Fleet 2 is away on assignment.
[2026-01-19 03:15:01] Fleet 2 returns at 2026-01-19 02:03:21.
[2026-01-19 03:15:01] Fleet 3 activated.
[2026-01-19 03:15:01] Fleet 3 is away on assignment.
[2026-01-19 03:15:01] Fleet 3 returns at 2026-01-19 02:09:22.
[2026-01-19 03:15:01] Fleet 4 activated.
[2026-01-19 03:15:01] Fleet 4 is away on assignment.
[2026-01-19 03:15:01] Fleet 4 returns at 2026-01-19 02:53:44.
[2026-01-19 03:15:01] Expedition mode manual, please make sure expedition fleet doesn't occupy noro6's ship and equipment
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\kcauto_custom\kcauto\__main__.py", line 26, in <module>
    kcauto_main()
  File "C:\kcauto_custom\kcauto\startup\kcauto_wrapper.py", line 23, in kcauto_main
    _.kcauto.start_kancolle()
  File "C:\kcauto_custom\kcauto\startup\kcauto.py", line 41, in start_kancolle
    kca_u.kca.start_kancolle()
  File "C:\kcauto_custom\kcauto\util\kca.py", line 164, in start_kancolle
    api.api.update_from_api({
  File "C:\kcauto_custom\kcauto\api\api_core.py", line 103, in update_from_api
    res = self._load_api_data(
          ^^^^^^^^^^^^^^^^^^^^
  File "C:\kcauto_custom\kcauto\api\api_core.py", line 142, in _load_api_data
    return self._process_port(data)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\kcauto_custom\kcauto\api\api_core.py", line 295, in _process_port
    flt.fleets.load_custom_fleets()
  File "C:\kcauto_custom\kcauto\fleet\fleet_core.py", line 215, in load_custom_fleets
    self.fleets = {**self.fleets, **self._noro6_to_kcauto()}
                                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\kcauto_custom\kcauto\fleet\fleet_core.py", line 424, in _noro6_to_kcauto
    Equipment name and production id: {[f'{eq.name} {eq.production_id}' for eq in ship.equipments]}, \
                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\kcauto_custom\kcauto\fleet\fleet_core.py", line 424, in <listcomp>
    Equipment name and production id: {[f'{eq.name} {eq.production_id}' for eq in ship.equipments]}, \
                                           ^^^^^^^
  File "C:\kcauto_custom\kcauto\ships\equipment.py", line 93, in name
    return self.static_data['api_name']
           ^^^^^^^^^^^^^^^^
  File "C:\kcauto_custom\kcauto\ships\equipment.py", line 63, in static_data
    return Equipment.equipment_static_data.get(self.model_id, None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'list' object has no attribute 'get'
Saving screenshots to 'crash_screenshots'...
Failed to save screenshot 0: OpenCV(4.11.0) D:\a\opencv-python\opencv-python\opencv\modules\imgcodecs\src\loadsave.cpp:929: error: (-215:Assertion failed) !_img.empty() in function 'cv::imwrite'

Failed to save screenshot 1: OpenCV(4.11.0) D:\a\opencv-python\opencv-python\opencv\modules\imgcodecs\src\loadsave.cpp:929: error: (-215:Assertion failed) !_img.empty() in function 'cv::imwrite'

Failed to save screenshot 2: OpenCV(4.11.0) D:\a\opencv-python\opencv-python\opencv\modules\imgcodecs\src\loadsave.cpp:929: error: (-215:Assertion failed) !_img.empty() in function 'cv::imwrite'

Failed to save screenshot 3: OpenCV(4.11.0) D:\a\opencv-python\opencv-python\opencv\modules\imgcodecs\src\loadsave.cpp:929: error: (-215:Assertion failed) !_img.empty() in function 'cv::imwrite'

Failed to save screenshot 4: OpenCV(4.11.0) D:\a\opencv-python\opencv-python\opencv\modules\imgcodecs\src\loadsave.cpp:929: error: (-215:Assertion failed) !_img.empty() in function 'cv::imwrite'

```

[2026-01-19-03-14-26.log](https://github.com/user-attachments/files/24697574/2026-01-19-03-14-26.log)



#### Config
Here is my config file:

[config_cui.json](https://github.com/user-attachments/files/24697580/config_cui.json)

--- a/data/equipment/ignore_equipment.json
+++ b/data/equipment/ignore_equipment.json
@@ -1,0 +1,3 @@
+{
+    "equipment_production_id": []
+}

--- a/kcauto/api/api_core.py
+++ b/kcauto/api/api_core.py
@@ -250,7 +250,7 @@ class ApiWrapper(object):
 
     def _process_require_info(self, data):
         try:
-            JsonData.dump_json(data['api_data']['api_slot_item'], 'data|temp|equipment_list.json')
+            JsonData.dump_json(self._filter_equipment(data['api_data']['api_slot_item']), 'data|temp|equipment_list.json')
         except KeyError:
             Log.log_debug("No equipment found in API response.")
             
@@ -461,7 +461,7 @@ class ApiWrapper(object):
 
     def _process_equipment_data(self, data):
         try:
-            JsonData.dump_json(data['api_data'], 'data|temp|equipment_list.json')
+            JsonData.dump_json(self._filter_equipment(data['api_data']), 'data|temp|equipment_list.json')
         except KeyError:
             Log.log_debug("No equipment found in API response.")
 
@@ -537,5 +537,32 @@ class ApiWrapper(object):
             
         except KeyError:
             Log.log_debug("No provisional equipment data found in API response")
+            
+    def _filter_equipment(self, equipment_data : list):
+        """Filter equipment based on ignore_equipment.json
+            This is a workaround to ignore certain equipment from being processed, 
+            due to Kancolle API providing ghost equipment entries that do not exist in the game.
+            See [record on Github](https://github.com/XVs32/kcauto_custom/discussions/123#discussioncomment-13599782) for more details.
+            args:
+                equipment_data (list): List of equipment data from API, ex.[{"api_id": 1, "api_slotitem_id": 33, "api_locked": 0, "api_level": 0},...]
+            returns:
+                filtered_equipment_data (list): List of equipment data after filtering
+        """
+         
+        try:
+            ignore_equipment_data = JsonData.load_json('data|equipment|ignore_equipment.json')
+        except FileNotFoundError:
+            ignore_equipment_data = {"equipment_production_id": []}
+            JsonData.dump_json(ignore_equipment_data, 'data|equipment|ignore_equipment.json')
+        ignored_equipment_production_id = ignore_equipment_data["equipment_production_id"]
+        
+        for i in range(len(equipment_data) -1 , -1, -1):
+            if equipment_data[i]['api_id'] in ignored_equipment_production_id:
+                Log.log_debug(f"Ignoring equipment with production id {equipment_data[i]['api_id']}")
+                equipment_data.pop(i)
+        
+        return equipment_data
+        
+            
 
 api = ApiWrapper()

